### PR TITLE
Improve main README

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,67 @@
 # CosmJS
 
-This is a JavaScript/TypeScript client-side binding to [wasmd](https://github.com/cosmwasm/wasmd), a sample blockchain for the [cosmwasm](https://github.com/confio/cosmwasm) smart contracting platform.
+CosmJS is the swiss army knife to power JavaScript based client solutions
```
Comment (willclarktech) [inline README.md:3]: "Swiss Army"

File: README.md [hunk 2]
```diff
@@ -1,12 +1,67 @@
...
+ranging from Web apps/explorers over browser extensions to server-side clients
+like faucets/scrapers in the Cosmos ecosystem.
+
+"Cosm" stands for Cosmos and "JS" stands for _runs everywhere_ – we actually
```
Comment (willclarktech) [inline README.md:7]: Suggest "is short for" instead of "stands for".

File: README.md [hunk 3]
```diff
@@ -1,12 +1,67 @@
...
+### Modularity
+
+We're pretty proud of the modularity and a clean dependency tree in this
+monorepo. This ensures software quality on our side and lets users pick exactly
+what they need. Here you see how everything fits together (every item is a npm
+package; right depends on left):
+
+![CosmJS dependency tree](docs/cosmjs-tree.png)
+
+<!--
```
Comment (willclarktech) [inline README.md:34]: Is this a GitHub feature?

File: README.md [hunk 4]
```diff
@@ -1,12 +1,67 @@
...
+### Supported JS environments
+
+Currently the codebase supports the following runtime environments:
+
+1. Node.js 10+
+2. Modern browsers (Chromium/Firefox/Safari, no Internet Explorer or
+   [Edge Spartan](https://en.wikipedia.org/wiki/Microsoft_Edge#Development))
+3. Browser extensions (Chromium/Firefox)
+
+Our current JavaScript target standard is ES2017, giving us native async/wait
```
Comment (willclarktech) [inline README.md:51]: "async/await"

File: README.md [hunk 5]
```diff
@@ -1,12 +1,67 @@
...
+support. We use WebAssembly to implement certain crypto.
```
Comment (willclarktech) [inline README.md:52]: "cryptographic functions"?

File: README.md [hunk 6]
```diff
@@ -1,12 +1,67 @@
...
+
+We're happy to adjust this list according to users' needs as long as you don't
+ask for Internet Explorer support. If your environment does not support Wasm, we
```
Comment (willclarktech) [inline README.md:55]: 🤣

File: README.md (hunk 2 above)
Comment (webmaster128) [reply README.md:7]: Thanks. What about the second "stands for"? Is this correct or a stange German translation? "is short for" probably does not work for the second case, right?

File: README.md (hunk 3 above)
Comment (webmaster128) [reply README.md:34]: A feature of markdown is and has always been the ability to embed HTML: https://daringfireball.net/projects/markdown/syntax#html

File: README.md (hunk 2 above)
Comment (willclarktech) [reply README.md:7]: Sounds natural to me 👍 .

**Comment by webmaster128** [comment]
Feedback implemented in the last commit

File: README.md (hunk 3 above)
Comment (willclarktech) [reply README.md:34]: Ah, I meant the content of the comment, not the comment itself. I thought for a minute that GitHub was doing some magic and generating an image based on your command.

Comment (webmaster128) [reply README.md:34]: Ah no, but it would be nice if they would. We could have a bot for rendering and checking-in the images.

**Pull request merged**